### DISCUSSION TITLE
Issues with type inference after upgrading to Prisma 5.4.2

### DISCUSSION THREAD

**Discussion by tmcallister92** [prisma/prisma#22518] [Q&A]
### Question

Hey all, 

Recently just embarked on the upgrade to Prisma 5.4.2 from Prisma 4.8.1.

Our setup comprises the following: 

1. TurboRebo Monorepo using PNPM 
2. A DB package that exports the Prisma Client.
3. TS 4.9.5

Since upgrading we are unable to use the inferred types from the Prisma Client, and are getting the following error: 

<img width="983" alt="image" src="https://github.com/prisma/prisma/assets/23154605/28f0a8d9-97cb-4db8-98ae-fd110fc6a974">

We have found multiple workarounds for this issue, the main one being explicitly typing the return type of the function using the generated type exported by the Prisma client, but would really like to be able to use type inference. 

The following github thread, documents this behavior in TS more thoroughly 

### How to reproduce (optional)

1. Monorepo with a db Package that exports the prisma client. 
2. application code that imports the prisma client.
3. function in application code that returns a type from the prisma client without explicitly typing it

### Expected behavior (optional)

I expect the type inference to work correctly without explicitly typing the return of a given function that returns a Prisma client call. 

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOs / Linux
- Database:  Postgres 16.1
- Node.js version:  18.12.0
- TS version: 4.9.5
- Prisma 5.4.2

### Additional info from TS Open Issue

https://github.com/microsoft/TypeScript/issues/47663#issuecomment-1519138189

**Comment by ludralph** [comment]
Hi @tmcallister92 👋 

Can you please confirm if this issue persist in a more recent version of Prisma, say Prisma version 5.8.0?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
